lib/expressions/Term.ts

```diff
@@ -91,38 +96,57 @@ export class Literal<T> extends Term {
 }
 
 export class NumericLiteral extends Literal<number> {
-  private static readonly specificFormatters: {[key in C.PrimitiveNumericType]: (val: number) => string } = {
+  private readonly specificFormatter: ((val: number) => string);
+  public constructor(
+    public typedValue: number,
+    public typeURL: RDF.NamedNode,
+    public strValue?: string,
+    public language?: string,
```

Comment:
Am I wrong to think we want to make strValue mandatory? If you give a literal to sparqlee and ask it back immediately. Shouldn't we get the same thing? This is particularly interesting for all types date related. If you where to give a date without a time zone, you wouldn't want to get one back with a arbitrary timezone? Wouldn't it pay of to generalize this?

Reply:
This is optional because we don't format prematurely when calculating, as the value could be changed many times, and we only care for formatting when exporting.

If you simply pass a literal from somewhere else and request that, strValue should be populated, and will be returned. See the `str` function. 

Anyway, this is only for performance, you need to have correct formatting in any case.

Datetime is an exception because there are no functions that manipulate the (typed) value https://www.w3.org/TR/sparql11-query/#func-date-time .

Might be worth adding this to some docs.

Reply:
Alright, thanks for the clarification.